File: solutions/py/7 kyu/TheFuscFunctionPartOne.py
#!/usr/bin/python3

# https://www.codewars.com/kata/570409d3d80ec699af001bf9/
def fusc(n):
    return n if n < 2 else fusc(n // 2) + fusc(n // 2 + 1) if n & 1 else fusc(n // 2)

File: solutions/scala/7 kyu/TheFuscFunctionPartOne.scala
//https://www.codewars.com/kata/570409d3d80ec699af001bf9/
object Sol {
  def fusc(n: Int): Int = {
    n match {
      case _ if n < 2 => n
      case _ if (n & 1) == 0 => fusc(n / 2)
      case _ => fusc(n / 2) + fusc(n / 2 + 1)
    }
  }
}